sound/soc/sof/topology.c

```diff
@@ -1812,61 +1765,83 @@ static int sof_process_load(struct snd_soc_component *scomp, int index,
 			       ARRAY_SIZE(comp_tokens), private->array,
 			       le32_to_cpu(private->size));
 	if (ret != 0) {
-		dev_err(sdev->dev, "error: parse process.cfg tokens failed %d\n",
+		dev_err(sdev->dev, "error: parse process tokens failed %d\n",
 			le32_to_cpu(private->size));
 		goto err;
 	}
 
 	sof_dbg_comp_config(scomp, &process->config);
 
+	process->size = 0;
+	swidget->private = process;
+
+	ret = sof_ipc_tx_message(sdev->ipc, process->comp.hdr.cmd, process,
+				 ipc_size, r, sizeof(*r));
+
+	if (ret < 0) {
+		dev_err(sdev->dev, "error: failed to create process comp\n");
+		goto err;
+	}
+
 	/*
-	 * found private data in control, so copy it.
-	 * get possible component controls - get size of all pdata,
-	 * then memcpy with headers
+	 * Get possible binary controls, which might have private data.
+	 * Then parse the size of the data and send it to dsp.
 	 */
 	for (i = 0; i < widget->num_kcontrols; i++) {
 		kc = &widget->kcontrol_news[i];
 
+		/* we only care about bytes control */
```

Comment:
why? Isn't that a restriction compared to the previous solution?

Reply:
@plbossart currently we are using component related control's binary blob to convey binary parameters to "process" component. and yeah, the name was not my idea...

As @ranj063 pointed out in another comment, why to even try to send other control types like enums or volume data at this point because it doens't make sense? Hence, I only try to find binary and "continue" with other types, if someone has tried to push that from topology, for some strange reason.